storage/aptosdb/src/pruner/ledger_pruner/event_store_pruner.rs

```diff
@@ -24,13 +30,32 @@ impl DBSubPruner for EventStorePruner {
 
     fn prune(&self, current_progress: Version, target_version: Version) -> Result<()> {
         let batch = SchemaBatch::new();
-        self.ledger_db
-            .event_db()
-            .prune_events(current_progress, target_version, &batch)?;
+        let indexer_deletes = SchemaBatch::new();
+        let event_indexer_enabled = self.internal_indexer_db.is_some()
+            && self.internal_indexer_db.as_ref().unwrap().event_enabled();
+        self.ledger_db.event_db().prune_events(
+            current_progress,
+            target_version,
+            &batch,
+            &indexer_deletes,
+            self.ledger_db.enable_storage_sharding(),
+            event_indexer_enabled,
+        )?;
         batch.put::<DbMetadataSchema>(
             &DbMetadataKey::EventPrunerProgress,
             &DbMetadataValue::Version(target_version),
         )?;
+        if let Some(indexer_db) = &self.internal_indexer_db {
+            if indexer_db.event_enabled() {
+                indexer_deletes.put::<InternalIndexerMetadataSchema>(
+                    &IndexerMetadataKey::EventPrunerProgress,
```

Comment:
Not used now since the APIs are trying to get the latest data. recorded here for debugging and diagnose to verify if indexer pruner is working as expected